src/Simulation/Simulators/QuantumSimulator/AssertProb.cs

```diff
@@ -46,7 +49,18 @@ public QSimAssertProb(QuantumSimulator m) : base(m)
 
                 if (Abs(ensemblePr - expectedPr) > tol)
                 {
-                    var extendedMsg = $"{msg}\n\tExpected:\t{expectedPr}\n\tActual:\t{ensemblePr}";
+                    string extendedMsg;
+                    if (result == Result.Zero)
+                    {
+                        // To account for the modification of expectedPr to (1 - expectedPr) when (result == Result.Zero), 
+                        // we must also update the ensemblePr to (1 - ensemblePr) when reporting the failure.
+                        extendedMsg = $"{msg}\n\tExpected:\t{expectedPr}\n\tActual:\t{(1 - ensemblePr)}";
```

Comment:
Now I may be confused, but shouldn't this be originalExpectedPr for the expected value here as well?
```suggestion
                        extendedMsg = $"{msg}\n\tExpected:\t{originalExpectedPr}\n\tActual:\t{(1 - ensemblePr)}";
```